homeassistant/components/remote/__init__.py

```diff
@@ -80,65 +82,69 @@ def send_command(hass, device, command, entity_id=None):
     hass.services.call(DOMAIN, SERVICE_SEND_COMMAND, data)
 
 
-def sync(hass, entity_id=None):
-    """Sync remote device."""
-    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
-    hass.services.call(DOMAIN, SERVICE_SYNC, data)
-
-
-def setup(hass, config):
+@asyncio.coroutine
+def async_setup(hass, config):
     """Track states and offer events for remotes."""
     component = EntityComponent(
         _LOGGER, DOMAIN, hass, SCAN_INTERVAL, GROUP_NAME_ALL_REMOTES)
-    component.setup(config)
+    yield from component.async_setup(config)
 
-    def handle_remote_service(service):
+    @asyncio.coroutine
+    def async_handle_remote_service(service):
         """Handle calls to the remote services."""
-        target_remotes = component.extract_from_service(service)
+        target_remotes = component.async_extract_from_service(service)
```

Comment:
I appreciate you porting everything over to async, it's still above my head!
I put a few extra logging lines in and finally realized that target entity_id's are not getting parsed correctly

`        _LOGGER.warning("service called for remotes: %s", target_remotes)`

Results in 

`16-12-03 19:16:09 WARNING (MainThread) [homeassistant.components.remote] service called for remotes: []
`